Source/Core/VideoBackends/Vulkan/StagingBuffer.cpp

```diff
@@ -180,59 +151,60 @@ bool StagingBuffer::AllocateBuffer(STAGING_BUFFER_TYPE type, VkDeviceSize size,
       0,                                     // uint32_t               queueFamilyIndexCount
       nullptr                                // const uint32_t*        pQueueFamilyIndices
   };
-  VkResult res =
-      vkCreateBuffer(g_vulkan_context->GetDevice(), &buffer_create_info, nullptr, out_buffer);
-  if (res != VK_SUCCESS)
+
+  VmaAllocationCreateInfo alloc_create_info = {};
+  alloc_create_info.flags =
+      VMA_ALLOCATION_CREATE_MAPPED_BIT | VMA_ALLOCATION_CREATE_WITHIN_BUDGET_BIT;
+  alloc_create_info.usage = VMA_MEMORY_USAGE_AUTO;
+  alloc_create_info.pool = VK_NULL_HANDLE;
+  alloc_create_info.pUserData = nullptr;
+  alloc_create_info.priority = 0.0;
+  alloc_create_info.preferredFlags = 0;
+  alloc_create_info.requiredFlags = 0;
+
+  if (DriverDetails::HasBug(DriverDetails::BUG_SLOW_CACHED_READBACK_MEMORY)) [[unlikely]]
   {
-    LOG_VULKAN_ERROR(res, "vkCreateBuffer failed: ");
-    return false;
+    // Prefer COHERENT over CACHED
+    alloc_create_info.usage = VMA_MEMORY_USAGE_UNKNOWN;
+    alloc_create_info.requiredFlags =
+        VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT;
+    alloc_create_info.preferredFlags = VK_MEMORY_PROPERTY_HOST_CACHED_BIT;
```

Comment:
Is the comment here correct? It looks like this requires COHERENT, and wants CACHED if it's available but is OK with it not being CACHED. I would assume that in this setup it's not possible to get CACHED but not COHERENT.

Reply:
When it's not possible to get both at the same time, we'd rather have COHERENT than CACHED. So it prefers COHERENT over CACHED.

Reply:
Maybe `// Prefer COHERENT + CACHED, but if that's not possible use just COHERENT`?